musicdroid/src/main/java/org/catrobat/musicdroid/pocketmusic/note/midi/ProjectToMidiConverter.java

```diff
@@ -55,12 +55,34 @@
 
 	public ProjectToMidiConverter() {
 		eventConverter = new NoteEventToMidiEventConverter();
-		usedChannels = new ArrayList<MusicalInstrument>();
+		usedChannels = new ArrayList<>();
 	}
-    
-	public void writeProjectAsMidi(Project project) throws IOException, MidiException {
-		MidiFile midi = convertProject(project);
 
+    public boolean deleteMidiByName(String name)
+    {
+        File file = new File(ProjectToMidiConverter.MIDI_FOLDER, name + ProjectToMidiConverter.MIDI_FILE_EXTENSION);
+
+        if(file.delete())
+            return true;
+
+        return false;
+    }
+
+    public void writeProjectAsMidi(Project project) throws IOException, MidiException {
+        MidiFile midiFile = convertProject(project);
+
+        checkMidiFolder();
+
+        File file = getMidiFileFromProjectName(project.getName());
+
+        if (file.exists()) {
+            throw new IOException("Project " + project.getName() + "already exists");
+        }
```

Comment:
Should not be in code but in external file for texts because we want to be able to translate our texts sometimes